index.js

```diff
@@ -30,60 +77,85 @@ const checkVersion = (engineName, command) => {
     // eslint-disable-next-line global-require, import/no-dynamic-require
     pkg = require(pkgJsonPath);
   } catch (e) {
-    console.log(`Unable to find ${pkgJsonPath}!  😱`);
-    console.log('');
-    console.log(
-      'Please ensure that this script is executed in the same directory.',
+    console.error(`Unable to find ${pkgJsonPath}!  😱${EOL}`);
+    console.error(
+      `Please ensure that this script is executed in the same directory.${EOL}`,
     );
-    console.log('');
     process.exit(1);
   }
 
-  let expected = null;
-
-  try {
-    expected = pkg.engines[engineName];
-  } catch (e) {
+  if (!pkg.engines || !pkg.engines[engineName]) {
     console.log(
-      `There is no engine named ${engineName} specified in package.json!  😱`,
+      `There is no engine named "${engineName}" specified in package.json!  😱${EOL}`,
     );
-    console.log('');
     process.exit(1);
   }
 
-  let version = null;
+  const expected = pkg.engines[engineName];
+
+  let globalVersion = null;
+  let localVersion = null;
+  let usedVersion = null;
+  let globalVersionValid = null;
+  let localVersionValid = null;
 
   try {
-    version = execSync(command)
+    const localBinFolder = execSync('npm bin')
       .toString()
-      .replace('v', '')
       .trim();
+
+    const hasLocalVersion = existsSync(
+      `${localBinFolder}${path.sep}${engineName}`,
+    );
+
+    globalVersion = getVersion({ command, localBinFolder });
+    globalVersionValid = semver.satisfies(globalVersion, expected);
+
+    console.log(`Expected ${engineName} version:\t${expected}`);
+    console.log(
+      `╰─ Global ${engineName} version:\t${globalVersion}\t${
+        globalVersionValid ? '✅️' : '❌️'
```

Comment:
If the script errors, I think it should also output something like `for more information, run this script again with the verbose flag (-v)` as this saves users from having to come and find the documentation 👍